packages/remix-react/components.tsx

```diff
@@ -1490,40 +881,13 @@ export interface RouteMatch {
   handle: undefined | { [key: string]: any };
 }
 
-/**
- * Returns the current route matches on the page. This is useful for creating
- * layout abstractions with your current routes.
- *
- * @see https://remix.run/api/remix#usematches
- */
-export function useMatches(): RouteMatch[] {
-  let { matches, routeData, routeModules } = useRemixEntryContext();
-
-  return React.useMemo(
-    () =>
-      matches.map((match) => {
-        let { pathname, params } = match;
-        return {
-          id: match.route.id,
-          pathname,
-          params,
-          data: routeData[match.route.id],
-          // if the module fails to load or an error/response is thrown, the module
-          // won't be defined.
-          handle: routeModules[match.route.id]?.handle,
-        };
-      }),
-    [matches, routeData, routeModules]
-  );
-}
-
 /**
  * Returns the JSON parsed data from the current route's `loader`.
  *
  * @see https://remix.run/api/remix#useloaderdata
  */
 export function useLoaderData<T = AppData>(): SerializeFrom<T> {
-  return useRemixRouteContext().data;
+  return useLoaderDataRR() as SerializeFrom<T>;
```

Comment:
This is the current solution until we get proper generics in RR - then we can re-export directly.  It felt like that was a whole undertaking unto itself though :)